src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -138,7 +147,7 @@ public void deleteRoles(final Node node) throws RepositoryException {
         // Read the effective Acl and return a list of acl:Authorization statements
         final List<WebACAuthorization> authorizations = effectiveAcl
                 .map(uncheck(x -> getAuthorizations(x.getLeft().toString())))
-                .orElse(new ArrayList<>());
+                .orElseGet(() -> getDefaultAuthorizations());
```

Comment:
Because `getEffectiveAcl` returns a `Pair<URI, FedoraResource>` which than gets loaded and the children (e.g. acl:Authorization nodes) are read -- and all of that is irrelevant for a default acl (which is already an acl:Authorization).

That is, I don't know what a filesystem-based `FedoraResource` would be (?). If we were to go that direction, the `getAuthorizations` method would need to be completely rewritten. This route is extremely light-weight -- the default is loaded only if an in-domain acl resource is not found.